lls/nd.c

```diff
@@ -647,8 +653,14 @@ process_nd(struct lls_config *lls_conf, struct gatekeeper_if *iface,
 	}
 
 	ipv6_hdr = (struct ipv6_hdr *)&eth_hdr[1];
-	icmpv6_hdr = (struct icmpv6_hdr *)&ipv6_hdr[1];
-	icmpv6_len = pkt_len - (sizeof(*eth_hdr) + sizeof(*ipv6_hdr));
+	start = ipv6_skip_exthdr(ipv6_hdr, remaining_len,
+		sizeof(*ipv6_hdr), &nexthdr, &frag_off);
+	if (start < 0 || nexthdr != IPPROTO_ICMPV6)
+		return -1;
+
+	icmpv6_hdr = (struct icmpv6_hdr *)((uint8_t *)ipv6_hdr + start);
+	icmpv6_len = pkt_len - (sizeof(struct ether_hdr) +
+		sizeof(*ipv6_hdr) + start);
```

Comment:
Isn't `start` the number of bytes between the start of the IPv6 header and the first byte after any extension headers? If so, I think there is a bug in this assignment statement, since `start` would be double counting the bytes already counted by `sizeof(*ipv6_hdr)`.